### PR TITLE
Change chat status to finished

### PR DESCRIPTION
**Pull request by guilleLara**
endpoint created to finish a chat, see all documentation at README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,10 @@
-# pis2019-chat-dashboard
\ No newline at end of file
+# pis2019-chat-dashboard
```
Comment (petterboussard) [inline README.md]: This line should be removed.

File: README.md [hunk 2]
```diff
@@ -1 +1,10 @@
...
+
+#### Change chat state to FINISHED
+PUT     {app_domain}/api/v1/chats/:chat_id/finish
+
+**Headers**:
+* Content-type: application/json
+
+**example, if you want to finish a chat with id = 5, you must be logged in and send:**
+{app_domain}/api/v1/chats/5/finish
```
Comment (petterboussard) [inline README.md]: Please make a decent documentation of your endpoint, the front-end team will appreciate it! :)

File: app/views/api/v1/chats/_chat.json.jbuilder [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
```
Comment (petterboussard) [inline app/views/api/v1/chats/_chat.json.jbuilder]: Why is this file necessary?

File: spec/factories/user_factory.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :user do
+    # provider { Faker::String.random }
```
Comment (petterboussard) [inline spec/factories/user_factory.rb]: Please get rid of comments.

File: app/controllers/api/v1/chats_controller.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  before_action :authenticate_user!, only: [:finish]
+
+  def finish
+    chat = Chat.find(params[:chat_id])
+    chat.state = 2 # finished
+    if chat.save
+      render json: chat, status: :ok
+    else
+      render json: chat.errors.messages, status: :not_modified
+    end
```
Comment (agupirotto) [inline app/controllers/api/v1/chats_controller.rb]: This is a better way to do the same by following some good Ruby practices.

return render json: chat, status: :ok if chat.save
render json: chat.errors.messages, status: :not_modified

Comment (guilleLara) [reply app/controllers/api/v1/chats_controller.rb]: I understand you, but it seemed more legible to me doing the if - else statement in that way. Also it helps to understand the tests for the method

File: app/controllers/api/v1/chats_controller.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  rescue ActiveRecord::RecordNotFound
```
Comment (jicc96) [inline app/controllers/api/v1/chats_controller.rb]: This could be verified in the code why let it throw an exception for something easily verifiable?

File: spec/requests/chats_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+      it 'should respond with error bad request if chat_id is invalid' do
+        login
+        auth_params = get_auth_params_from_login_response_headers(response)
+        put api_v1_chat_finish_path(-1), headers: auth_params
+        expect(response).to have_http_status(400)
+      end
+    end
+  end
+
+  it 'should respond with success status' do
```
Comment (jicc96) [inline spec/requests/chats_controller_spec.rb]: This test looks like it should be inside the describe context above it, there is a reason why it is not?

File: app/controllers/api/v1/chats_controller.rb (hunk 6 above)
Comment (guilleLara) [reply app/controllers/api/v1/chats_controller.rb]: the exception is correctly catched with rescue and returns an appropiate message. Petter told me it was a good practice to do it in that way. Do you think adding an 'if chat.nil?' statement is cleaner?

File: spec/requests/chats_controller_spec.rb (hunk 7 above)
Comment (guilleLara) [reply spec/requests/chats_controller_spec.rb]: It is not in the context of the chat controller tests above. The test is verifying that the user is correctly logged in (using the method POST /login) before accesing the chat controller method.

File: app/controllers/api/v1/chats_controller.rb [hunk 8]
```diff
@@ -1,7 +1,19 @@
 # frozen_string_literal: true
 
 class Api::V1::ChatsController < ApplicationController
-  before_action :authenticate_user!, only: [:new_dashboard_message]
+  before_action :authenticate_user!, only: %i[finish new_dashboard_message]
+
+  def finish
```
Comment (jicc96) [inline app/controllers/api/v1/chats_controller.rb]: You should verify that the logged user is assigned to the chat, and if that user is the active sender for the chat.
